### PR TITLE
[hist] fix missing default parameters in save-as

### PR DESCRIPTION
**Pull request by ferdymercury**
# This Pull request:

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

Fixes https://github.com/root-project/root/issues/14863

### CODE REVIEW

**Review by vepadulano — CHANGES_REQUESTED** [review]
Thank you @ferdymercury ! I also pushed a change in the test to make sure the case highlighted by the related issue is tested

File: hist/hist/inc/TGraph.h [hunk 1]
```diff
@@ -178,7 +178,7 @@ class TGraph : public TNamed, public TAttLine, public TAttFill, public TAttMarke
    virtual Int_t         RemovePoint(); // *MENU*
    virtual Int_t         RemovePoint(Int_t ipoint);
    void                  SavePrimitive(std::ostream &out, Option_t *option = "") override;
-   void                  SaveAs(const char *filename, Option_t *option = "") const override; // *MENU*
+   void                  SaveAs(const char *filename = "graph", Option_t *option = "") const override; // *MENU*
```
Comment (vepadulano) [inline hist/hist/inc/TGraph.h:181]: Can we keep the changes of this PR related to `TH1` only so we address the related issue with more focus? Since this default argument was simply untested, we would need some extra tests also for `TGraph`, but I think this should go in another PR

Comment (ferdymercury) [reply hist/hist/inc/TGraph.h]: yep!

**Review by vepadulano — APPROVED** [review]
LGMT, thank you so much for the prompt support @ferdymercury ! I will wait for the CI to confirm the changes work and then merge
